### DISCUSSION TITLE
Support for standalone component (the recommended way by Angular 17)

### DISCUSSION THREAD

**Discussion by DABBEL-cheng** [primefaces/community#506] [PrimeNG Templates] [answered]
My company bought Apollo template, however, when I check the code, I found it's not following the best practice of Angular 17. For example, Angular document says

> The Angular team recommends using standalone components for all new development.

And the code is still using the Angular 16 and `NgModule` way. Is there a plan to update the code to Angular 17 and follow its recommended way of using standalone component? 

Screenshot from [Angular document](https://angular.dev/guide/components/importing)
![image](https://github.com/primefaces/community/assets/110891638/0bd9d89d-b6f8-4773-8352-e8cada7c0f32)

**Comment by mehmetcetin01140** [comment, marked as answer]
Hi,
We are planning to migrate our templates to angular 17 soon. We will evaluate best practices.

Thanks for feedback.

**Reply by DABBEL-cheng** [reply]
Thank you for the prompt reply. Great to hear about the plan of migrating to the newest version as well as evaluating best practices.

**Comment by EroNWs** [comment]
When do you plan to do this upgrade? Can I get information? It is important!

**Comment by armingdev** [comment]
Any updates on how we are standing on standalone components? Any ETA for this? It will have a big impact on performances. (not asking about any template but primeng components general)

**Comment by cetincakiroglu** [comment]
Hi,

We're currently focused on passthrough, unstyled mode and new styled mode features. We're planning to update library as standalone components afterward.

**Reply by nchigwamba** [reply]
Hi,

Is there a timeline for this?

**Reply by cetincakiroglu** [reply]
Hi,

There isn't an exact timeline for this task. We need to implement passthrough, unstyled, new styled mode beforehand. But it would happen between Angular v18-v20.
